src/base/Quantity.hh

```diff
@@ -48,20 +53,28 @@ struct UnitlessQuantity
  * A relativistic equation that operates on these quantities can do so without
  * unnecessary floating point operations involving the speed of light:
  * \code
-   real_type eval = energy.value(); // Natural units
-   MevMomentum momentum{std::sqrt(eval * eval + 2 * mass.value() * eval)};
+   real_type eval = qvalue<MevEnergy>(energy); // Natural units
+   MevMomentum momentum{std::sqrt(eval * eval
+                                  + 2 * qvalue<MevMass>(mass) * eval)};
    \endcode
  * The resulting quantity can be converted to the native Celeritas unit system
- * with a `unit_cast`, which multiplies in the constant value of
+ * with `from_quantity`, which multiplies in the constant value of
  * ElMomentumUnit:
  * \code
- * real_type mom = unit_cast(momentum);
+ * real_type mom = from_quantity(momentum);
  * \endcode
  *
+ * When using a Quantity from another part of the code, e.g. an imported unit
+ * system, use the \c quantity free function rather than \c .value() in order
+ * to guarantee consistency of units between source and destination.
+ *
  * \note The Quantity is designed to be a simple "strong type" class, not a
- * complex mathematical class. To operate on quantities, you must use `value()`
- * or `unit_cast` as appropriate and operate on the numeric values as
- * appropriate, then return a new Quantity class as appropriate.
+ * complex mathematical class. To operate on quantities, you must use
```

Comment:
At the moment the code seems to still be using `value` (which I think is a better alternative, especially since `quantity` and `Quantity` only differ by the casing).